programs/uxd/src/instructions/mercurial/collect_profit_of_mercurial_vault_depository.rs

```diff
@@ -0,0 +1,298 @@
+    // We accept theses losses as the money is still in the vault. We collect a bit less profit.
+    CollectProfitOfMercurialVaultDepository::check_redeemed_collateral_amount_to_match_target(
+        collateral_balance_change,
+        collectable_profits_value,
+        possible_lp_token_precision_loss_collateral_value,
+    )?;
+
+    // 8 - update accounting
+    let current_time_as_unix_timestamp = u64::try_from(Clock::get()?.unix_timestamp)
+        .ok()
+        .ok_or_else(|| error!(UxdError::MathError))?;
+
+    ctx.accounts
+        .depository
+        .load_mut()?
+        .update_onchain_accounting_following_profits_collection(
+            collateral_balance_change,
+            current_time_as_unix_timestamp,
+        )?;
+
+    ctx.accounts
+        .controller
+        .load_mut()?
+        .update_onchain_accounting_following_profit_collection(collateral_balance_change)?;
+
+    Ok(())
+}
+
+// Into functions
+impl<'info> CollectProfitOfMercurialVaultDepository<'info> {
+    pub fn into_withdraw_collateral_from_mercurial_vault_context(
+        &self,
+    ) -> CpiContext<
+        '_,
+        '_,
+        '_,
+        'info,
+        mercurial_vault::cpi::accounts::DepositWithdrawLiquidity<'info>,
+    > {
+        let cpi_accounts = mercurial_vault::cpi::accounts::DepositWithdrawLiquidity {
+            vault: self.mercurial_vault.to_account_info(),
+            token_vault: self.mercurial_vault_collateral_token_safe.to_account_info(),
+            lp_mint: self.mercurial_vault_lp_mint.to_account_info(),
+            user_token: self.profits_beneficiary_collateral.to_account_info(),
+            user_lp: self.depository_lp_token_vault.to_account_info(),
+            user: self.depository.to_account_info(),
+            token_program: self.token_program.to_account_info(),
+        };
+        let cpi_program = self.mercurial_vault_program.to_account_info();
+        CpiContext::new(cpi_program, cpi_accounts)
+    }
+}
+
+// Calculation/Check functions
+impl<'info> CollectProfitOfMercurialVaultDepository<'info> {
+    fn calculate_lp_token_amount_to_match_collectable_profits_value(
+        &self,
+        target_value: u64,
+    ) -> Result<u64> {
+        let current_time = u64::try_from(Clock::get()?.unix_timestamp)
+            .ok()
+            .ok_or_else(|| error!(UxdError::MathError))?;
+
+        // Because it's u64 type, we will never withdraw too much due to precision loss, but withdraw less.
+        // We withdraw less interests and fee due to precision loss and that's ok
+        self.mercurial_vault
+            .get_unmint_amount(
+                current_time,
+                target_value,
+                self.mercurial_vault_lp_mint.supply,
+            )
+            .ok_or_else(|| error!(UxdError::MathError))
+    }
+
+    pub fn calculate_collectable_profits_value(&self) -> Result<u64> {
+        let owned_lp_tokens_value = I80F48::checked_from_num(
```

Comment:
imo this code should not use I80F48, simple u64 or 128 should suffice here, as even the mercurial's function does not return a I80F48

Reply:
yes removed the usage of I80F48 on profit collection, it still got usage on rest of the mercurial ixs, let me remove them in this PR tgt

Reply:
Yeah no need to remove it on other IXs for now, just remove it in all modified files required for this PR